libraries/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -880,6 +959,14 @@ export class RushConfiguration {
         );
       }
       this._projectsByName.set(project.packageName, project);
+      if (projectJson.subspace) {
+        const subspaceName: string = projectJson.subspace;
+        if (this._subspaceProjectsCache.has(subspaceName)) {
+          (this._subspaceProjectsCache.get(subspaceName) as RushConfigurationProject[]).push(project);
+        } else {
+          this._subspaceProjectsCache.set(subspaceName, [project]);
+        }
```

Comment:
```suggestion
        const projectsForSubspace: RushConfigurationProject[] | undefined = this._subspaceProjectsCache.get(subspaceName);
        if (projectsForSubspace) {
          projectsForSubspace.push(project);
        } else {
          this._subspaceProjectsCache.set(subspaceName, [project]);
        }
```
Avoid the unnecessary double lookup in the success path.